pcs/lib/resource_agent/xml.py

```diff
@@ -137,20 +138,36 @@ def load_fake_agent_metadata(
 
 
 def parse_metadata(
-    name: ResourceAgentName, metadata: _Element
+    name: ResourceAgentName,
+    metadata: _Element,
+    report_processor: reports.ReportProcessor,
+    report_warnings=True,
 ) -> Union[ResourceAgentMetadataOcf1_0, ResourceAgentMetadataOcf1_1]:
     """
     Parse XML metadata to a dataclass
 
     name -- name of an agent
     metadata -- metadata XML document
+    report_processor
+    report_warnings -- to disable warnings not to pollute stdout with them
     """
     ocf_version = _get_ocf_version(metadata)
     if ocf_version == const.OCF_1_0:
         return _parse_agent_1_0(name, metadata)
     if ocf_version == const.OCF_1_1:
         return _parse_agent_1_1(name, metadata)
-    raise UnsupportedOcfVersion(name.full_name, ocf_version)
+    if report_warnings:
+        report_processor.report(
+            reports.ReportItem.warning(
+                reports.messages.AgentImplementsUnsupportedOcfVersionAssumedVersion(
+                    name.full_name,
+                    ocf_version,
+                    sorted(const.SUPPORTED_OCF_VERSIONS),
+                    const.OCF_1_0,
+                )
+            )
+        )
+    return _parse_agent_1_0(name, metadata)
```

Comment:
We should treat case of unsupported ocf version more carefully as there is no prior schema validation. It can end up with some unhandled exceptions caused by accessing non-existing attributes of a xml element. Therefore we should create a separate function similar to `_parse_agent_1_0` where missing attributes which are not vital for pcs to work can be ignored and raise an exception if it's not possible to continue.  Or another option would be to use `_parse_agent_1_0` as it is but catch any possible exception and raise our own exception if there was any problem. Either way, the exception produced by our code should then be handled by higher layers (printing out an error or skipping the agent).

Reply:
That's not really worth the effort. The purpose of this patch is to allow valid agents to be processed even if they report bad OCF version (due to version tag and version attribute confusion). Such agents are almost always processable by `_parse_agent_1_0`, which treats majority of elements and attributes as optional anyway. In the rare cases of an exception, we're fine with it. And it's caught in higher layers anyway.

Reply:
I understand the purpose of this patch. However, I still think pcs should not exit with a traceback in case agent's version is not supported and there is actually a mistake in agent's metadata (e.g. missing name attribute in parameter xml tag). This is not the case for agents with supported versions as their metadata are validated against xml schema.

I gave it a try and I added a parameter without name in test `pcs_test.tier0.lib.resource_agent.test_facade.ResourceAgentFacadeFactory.test_facade_bad_ocf_version` and it is actually failing with a `KeyError`.

Therefore we should either handle this exception somewhere or modify `pcs.lib.resource_agent.xml._metadata_xml_to_dom` in a way that unsupported metadata versions are also validated  with 1.0 schema (similarly how `pcs.lib.resource_agent.xml.parse_metadata` has been updated).

Reply:
Thinking about it little bit more, I can see a downside of ocf 1.0 validation in `pcs.lib.resource_agent.xml._metadata_xml_to_dom` for metadata with an unsupported version. In case it's not valid pcs will only print out error about wrong format of metadata (which is fine because pcs would fail later on anyway) but there will be no warning about unsupported metadata format version. But I believe this is not a big deal.

Reply:
I added a bunch of fixes to make pcs process OCF 1.0 agents in the same way as they were processed before resource agent parser overhaul.